collect_app/src/main/java/org/odk/collect/android/utilities/TextUtils.java

```diff
@@ -71,26 +71,32 @@ private TextUtils() {
 
     protected static String markdownToHtml(String text) {
 
-        text = text.replaceAll("<([^a-zA-Z/])", "&lt;$1");
+        text = text.replaceAll("<([^a-zA-Z/])", "/</gm,&lt;");
         // https://github.com/enketo/enketo-transformer/blob/master/src/markdown.js
 
         // span - replaced &lt; and &gt; with <>
-        text = ReplaceCallback.replace("(?s)<\\s?span([^\\/\n]*)>((?:(?!<\\/).)+)<\\/\\s?span\\s?>",
+        text = ReplaceCallback.replace("/&lt;\\s?span([^/\\n]*)&gt;((?:(?!&lt;\\/).)+)&lt;\\/\\s?span\\s?&gt;/gm, _createSpan >",
                 text, createSpan);
         // strong
-        text = text.replaceAll("(?s)__(.*?)__", "<strong>$1</strong>");
-        text = text.replaceAll("(?s)\\*\\*(.*?)\\*\\*", "<strong>$1</strong>");
+        text = text.replaceAll("/__(.*?)__/gm,", "<strong>$1</strong>");
+        text = text.replaceAll("/\\*\\*(.*?)\\*\\*/gm,", "<strong>$1</strong>");
         // emphasis
-        text = text.replaceAll("(?s)_([^\\s][^_\n]*)_", "<em>$1</em>");
-        text = text.replaceAll("(?s)\\*([^\\s][^\\*\n]*)\\*", "<em>$1</em>");
+        text = text.replaceAll("/_([^\\s][^_\\n]*)_/gm,", "<em>$1</em>");
+        text = text.replaceAll("/\\*([^\\s][^*\\n]*)\\*/gm,", "<em>$1</em>");
         // links
-        text = text.replaceAll("(?s)\\[([^\\]]*)\\]\\(([^\\)]+)\\)",
+        text = text.replaceAll("/\\[([^\\]]*)\\]\\(([^)]+)\\)/gm,",
                 "<a href=\"$2\" target=\"_blank\">$1</a>");
         // headers - requires ^ or breaks <font color="#f58a1f">color</font>
         text = ReplaceCallback.replace("(?s)^(#+)([^\n]*)$", text, createHeader);
         // paragraphs
         text = ReplaceCallback.replace("(?s)([^\n]+)\n", text, createParagraph);
 
+        // escape characters for * , _
```

Comment:
This comment doesn't match the code below!

Reply:
Yeah I have been working on the tests today and I have fixed every other test apart from the @Test
    public void markDownToHtml_EscapesLessThan() which is challenging. I will make a PR when all the tests work properly.

Reply:
Great! No need to make a new PR, you can simply push more commits to this same branch.